### PR TITLE
impl(docfx): support Doxygen `@page` documents

### PR DESCRIPTION
**Pull request by coryan**
This is enough to support the landing page for the common library. It is missing features, the plan of record is to add these as-needed.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10886)

### CODE REVIEW

File: docfx/doxygen2markdown.cc [hunk 1]
```diff
@@ -30,6 +30,95 @@ namespace {
...
+//   <xsd:attribute name="kind" type="DoxCompoundKind" />
+//   <xsd:attribute name="language" type="DoxLanguage" use="optional"/>
+//   <xsd:attribute name="prot" type="DoxProtectionKind" />
+//   <xsd:attribute name="final" type="DoxBool" use="optional"/>
+//   <xsd:attribute name="inline" type="DoxBool" use="optional"/>
+//   <xsd:attribute name="sealed" type="DoxBool" use="optional"/>
+//   <xsd:attribute name="abstract" type="DoxBool" use="optional"/>
+// </xsd:complexType>
+// clang-format on
+//
```
Comment (devbww) [inline docfx/doxygen2markdown.cc:89]: Typically you haven't added a blank comment line here.

File: docfx/doxygen2markdown.cc [hunk 2]
```diff
@@ -30,6 +30,95 @@ namespace {
 
 }  // namespace
 
+// A "page" appears in the generated XML as:
+// clang-format off
+//   <compounddef xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="indexpage" kind="page">
+// clang-format on
+//
+// That is, they are generic `compounddef` nodes, the same entity used to
```
Comment (devbww) [inline docfx/doxygen2markdown.cc:38]: s/nodes, the/nodes--the/ perhaps?

File: docfx/doxygen2markdown.cc [hunk 3]
```diff
@@ -158,6 +247,30 @@ bool AppendIfSect1(std::ostream& os, MarkdownContext const& ctx,
...
+//       <xsd:element name="internal" type="docInternalType" minOccurs="0" maxOccurs="unbounded"/>
+//       <xsd:element name="sect1" type="docSect1Type" minOccurs="0" maxOccurs="unbounded" />
+//     </xsd:sequence>
+//   </xsd:complexType>
+// clang-format on
+bool AppendIfDetailedDescription(std::ostream& os, MarkdownContext const& ctx,
+                                 pugi::xml_node const& node) {
+  if (std::string_view{node.name()} != "detaileddescription") return false;
+  for (auto const& child : node) {
+    // Unexpected: title, internal  -> we do not use this...
```
Comment (devbww) [inline docfx/doxygen2markdown.cc:266]: Two spaces again :-)

**Review by devbww — APPROVED** [review]

File: docfx/doxygen2markdown.cc (hunk 3 above)
Comment (coryan) [reply docfx/doxygen2markdown.cc]: Fixed.

File: docfx/doxygen2markdown.cc (hunk 2 above)
Comment (coryan) [reply docfx/doxygen2markdown.cc]: Fixed.

File: docfx/doxygen2markdown.cc (hunk 1 above)
Comment (coryan) [reply docfx/doxygen2markdown.cc]: Fixed and also fixed a few other inconsistent comments.

**Pull request merged**